Source/KakapoServer.swift

```diff
@@ -90,12 +92,18 @@ public final class KakapoServer: NSURLProtocol {
     }
     
     /// Start loading the matched requested, the route handler will be called and the returned object will be serialized.
-    override public func startLoading() {
-        KakapoServer.routers.filter { $0.canInitWithRequest(request) }.first!.startLoading(self)
+    override public func startLoading() {        
+        guard let router = KakapoServer.routers.filter({ $0.canInitWithRequest(request) }).first else {
+            return
+        }
+        router.startLoading(self)
     }
     
-    /// Not implemented yet, does nothing ATM. https://github.com/devlucky/Kakapo/issues/88
+    /// Stops the loading of the matched request.
     override public func stopLoading() {
-        /* TODO: implement stopLoading for delayed requests https://github.com/devlucky/Kakapo/issues/88 */
+        guard let router = KakapoServer.routers.filter({ $0.canInitWithRequest(request) }).first else {
```

Comment:
Actually those filterings + first! are a bad legacy implementation from our side on both `canInitWithRequest` and `startLoading` since this will always lead to O(N) time complexity in those methods (filter will always filter the whole array). 

The proper way to do it would be using `indexOf` which would save some good time in average case, could you adapt it?